app/assets/javascripts/discourse/app/routes/application.js

```diff
@@ -76,6 +76,11 @@ const ApplicationRoute = DiscourseRoute.extend({
       !this.restrictedRouting.isAllowedRoute(transition.to.name)
     ) {
       transition.abort();
+      this.router.transitionTo(
```

Comment:
I think it may be slightly better to do `replaceWith` ?  From https://guides.emberjs.com/release/routing/redirection/ : 

> The other one is [replaceWith()](https://api.emberjs.com/ember/release/classes/RouterService/methods/replaceWith?anchor=replaceWith) which works the same way as transitionTo(). The only difference between them is how they manage history. replaceWith() substitutes the current route entry and replaces it with that of the route we are redirecting to, while transitionTo() leaves the entry for the current route and creates a new one for the redirection.